integration/simulation/transaction_manager.go

```diff
@@ -1 +1,97 @@
 package simulation
+
+import (
+	"sync"
+	"time"
+
+	common2 "github.com/obscuronet/obscuro-playground/go/obscuronode/common"
+
+	"github.com/obscuronet/obscuro-playground/go/obscuronode/enclave"
+
+	"github.com/google/uuid"
+	"github.com/obscuronet/obscuro-playground/go/common"
+	wallet_mock "github.com/obscuronet/obscuro-playground/integration/walletmock"
+)
+
+type TransactionManager struct {
+	l1NetworkConfig    *L1NetworkCfg
+	l2NetworkConfig    *L2NetworkCfg
+	blockDuration      uint64
+	simulationTimeInUS int
+	stats              *Stats
+	wallets            []wallet_mock.Wallet
+	l1TransactionsLock sync.RWMutex
+	l1Transactions     common.Transactions
+	l2TransactionsLock sync.RWMutex
+	l2Transactions     enclave.Transactions
+}
+
+func (m *TransactionManager) Start(l1 *L1NetworkCfg, l2 *L2NetworkCfg, duration uint64, us int, stats *Stats) {
+	m.l1NetworkConfig = l1
+	m.l2NetworkConfig = l2
+	m.blockDuration = duration
+	m.simulationTimeInUS = us
+	m.stats = stats
+
+	// deposit some initial amount into every user
+	for _, u := range m.wallets {
+		tx := deposit(u, INITIAL_BALANCE)
+		t, _ := tx.Encode()
+		m.l1NetworkConfig.BroadcastTx(t)
+		m.stats.Deposit(INITIAL_BALANCE)
+		time.Sleep(common.Duration(m.blockDuration / 3))
+	}
+
+	// inject numbers of transactions proportional to the simulation time, such that they can be processed
+	go injectRandomDeposits(m.wallets, m.l1NetworkConfig, m.blockDuration, m.simulationTimeInUS, m.stats, m.TrackL1Tx)
```

Comment:
shouldn't these functions also be part of this object?

Reply:
Yep 👍